### ISSUE TITLE
ban-untagged-todo could be better

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#534]
Currently `ban-untagged-todo` allows following cases:
```
// TODO Improve calc engine (@djones)
export function calcValue(): number { }
// TODO Improve calc engine (#332)
export function calcValue(): number { }
```

But prevents such cases:
```
// TODO(bartlomieju) Improve calc engine
export function calcValue(): number { }
```

The latter case is prevalent in deno's codebase and I'd rather not add 100s of `@` to satify the linter; so I propose to allow username just after `TODO` and don't require `@`, eg. `// TODO(bartlomieju) Improve calc engine`

Ref https://github.com/denoland/deno/issues/7410

**Comment by magurotuna** [comment]
I think that makes sense. Then, will the allowed TODO style be expressed like `todo\(\S+\)` in regex?

FYI: the current `ban-untagged-todo` implementation doesn't allow your first examples actually, since we currently use regex like 

```rust
static TODO_RE: Lazy<Regex> =
    Lazy::new(|| Regex::new(r#"todo\((#|@)\S+\)"#).unwrap());
```

so `TODO` and `(@username)` must be neighbored. I'm not sure if this is intentional behavior though.

**Comment by bartlomieju** [comment]
@magurotuna good catch, my main concern is requirement for `@` which I believe is superficial.

**Comment by magurotuna** [comment]
Yeah I agree with you, atmark before username isn't really necessary, considering about current usage of them.
